frontend/src/store/modules/user.ts

```diff
@@ -52,31 +60,55 @@ export default {
     setUserRoles: (state: any, roles: { [key: string]: any } | null) => {
       state.roles = roles;
     },
+    setMetadataCallback: (
+      state: any,
+      callback: ((snapshot: firebase.database.DataSnapshot) => void) | null
+    ) => {
+      state.metadataCallback = callback;
+    },
+    setTimestamp: (state: any, timestamp: number) => {
+      state.timestamp = timestamp;
+    }
   },
   actions: {
     signOut() {
       firebase.auth().signOut();
     },
-    async handleUser(
-      { commit, dispatch }: ActionContext<any, any>,
+    async onAuthStateChanged(
+      { dispatch, commit, state, getters }: ActionContext<any, any>,
       user: firebase.User | null
     ) {
       commit("setCurrentUser", user);
-      if (user) {
-        await dispatch(
-          "updateUserRoles",
-          (await user.getIdTokenResult()).claims.roles
-        );
-      } else {
-        await dispatch("updateUserRoles", null);
+
+      if (state.metadataCallback && getters.metadataRef) {
+        commit("setTimestamp", null);
```

Comment:
I think setting it to null is not a part of this piece of code. Here you only remove the callback. The timestamp (refresh time) should be set to current time along with updating the roles after the user signs in.